staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -83,29 +83,82 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 }
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
-func (g *Cloud) NodeAddresses(_ context.Context, _ types.NodeName) ([]v1.NodeAddress, error) {
-	internalIP, err := metadata.Get("instance/network-interfaces/0/ip")
-	if err != nil {
-		return nil, fmt.Errorf("couldn't get internal IP: %v", err)
+func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v1.NodeAddress, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	instanceName := string(nodeName)
+
+	if g.useMetadataServer {
+
```

Comment:
Nit: mind avoiding excessive empty lines so that future readers can have a easier time reading through?
```suggestion
// Use metadata server if possible
```